### PR TITLE
Reuse AcqEra/ProcVer from the InputDataset when creating StoreResults

### LINKED ISSUE #9092: StoreResults dataset should preserve the original input dataset name
The original issue can be found here: https://its.cern.ch/jira/browse/CMSCOMPPR-4907

For a test workflow: https://cmsweb.cern.ch/reqmgr2/fetch?rid=sagarwal_EmbeddingRun2017B-TauTauFinalState-inputDoubleMu_94X_miniAOD-v2_190305_162744_2306
Input dataset: /EmbeddingRun2017B/TauTauFinalState-inputDoubleMu_94X_miniAOD-v2/USER
Output dataset: /EmbeddingRun2017B/StoreResults-inputDoubleMu_94X_miniAOD-v1/USER is the output dataset 
The `TauTauFinalState` in the original input dataset has been replaced by `StoreResults`. As a result, several different input datasets come out with the same output dataset:
https://cmsweb.cern.ch/reqmgr2/fetch?rid=sagarwal_EmbeddingRun2017B-TauTauFinalState-inputDoubleMu_94X_miniAOD-v2_190305_162744_2306
https://cmsweb.cern.ch/reqmgr2/fetch?rid=sagarwal_EmbeddingRun2017B-ElTauFinalState-inputDoubleMu_94X_miniAOD-v2_190305_162740_957

@amaltaro Can you please fix this? Ideally the output name for the above example workflow should be /EmbeddingRun2017B/StoreResults-TauTauFinalState-inputDoubleMu_94X_miniAOD-v1/USER

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #9092 
Instead of using the hard-wired "StoreResults" for the AcquisitionEra, reuse the same as the InputDataset field (unless AcqEra has a dash in it, then we use only the first part of it).
Same for ProcessingVersion, keep the version the same as the input dataset.

@ericvaandering do you remember whether there was any rational for having a hard-coded AcquisitionEra?
@thongonary any chance you could test this fix for me?

### CODE REVIEW

File: bin/createStoreResults.py [hunk 1]
```diff
@@ -106,8 +107,12 @@ def buildRequest(userDict):
     # Set PrepID according to the date and time
     newSchema["PrepID"] = "StoreResults-%s" % time.strftime("%d%m%y-%H%M%S")
     # Truncate the ProcessingString, otherwise it can be larger than allowed
-    _, primDset, procDset, _ = newSchema['InputDataset'].split("/")
-    newSchema["ProcessingString"] = procDset.split("-")[1:-1][0][:80]  # first 80 chars
+    primDset, procDset, _tier = newSchema['InputDataset'].split("/")
```
Comment (amaltaro) [inline bin/createStoreResults.py]: Just so I don't forget, I need to add a `[1:]` to the end here.

**Comment by amaltaro** [comment]
Besides the small bug fix. @thongonary please confirm whether those two changes is what you really want to have.

**Comment by thongonary** [comment]
@amaltaro So this is 1 of the test workflow:
https://cmsweb.cern.ch/reqmgr2/fetch?rid=sagarwal_EmbeddingRun2017E-TauTauFinalState-inputDoubleMu_94X_miniAOD-v2_190314_104155_5567

I see that input and output datasets have the same name. Does that even work?

**Comment by amaltaro** [comment]
Haaa, thanks for spotting it Thong. Whether it works or not, in theory yes, in practice it will be a mess.
I'll then prefix the ProcessingString with `StoreResults_`, that should make it unique.

**Comment by amaltaro** [comment]
Patch was updated. @thongonary and @sharad1126 can you please test it again?

**Comment by amaltaro** [comment]
@thongonary @sharad1126  did you manage to test it? Does it now provide the behaviour you needed? If so, I can then merge it.

**Comment by sharad1126** [comment]
@thongonary can you please have a look at the CMSCMOPPR-4907 and reply on the PR

**Comment by sharad1126** [comment]
@amaltaro it works as expected and I think we can merge the PR

**Comment by amaltaro** [comment]
Thanks, Sharad and Thong. Merging

**Pull request merged**